tests/unit/topology/test_graph.py

```diff
@@ -58,9 +64,36 @@ def dummy_topology():
     return create_dummy_topology()
 
 
-def test_initialize_graph(  # pylint: disable=unused-argument
-    dummy_topology, particle_database
+@pytest.mark.parametrize(
+    "initial_state, final_state",
+    [
+        (["J/psi(1S)"], ["gamma", "pi0", "pi0"]),
+        (["J/psi(1S)"], ["K+", "K-", "pi+", "pi-"]),
+        (["e+", "e-"], ["gamma", "pi-", "pi+"]),
+        (["e+", "e-"], ["K+", "K-", "pi+", "pi-"]),
+    ],
+)
+def test_permutations(
+    initial_state, final_state, dummy_topology, particle_database
 ):
+    initial_state_with_spins = _safe_set_spin_projections(
```

Comment:
Ok, that's downside in the commit based reviews. This answers my question from before. However I'm questioning the `particle` API now. So concretely can the API be changed so that new function can provide this functionality (combined) and calls to  `_safe_set_spin_projection` can be avoided?

Reply:
By new function, do you mean that `_safe_set_spin_projections` now takes a list? I'm not sure how `_safe_set_spin_projection can be avoided, because we want to allow the user to not specify spin projections in case all allowed spin projections should be considered.

Reply:
Well the calls made here are really similar to what the `initialize_graphs` function is doing internally. So I'm saying, maybe instead `initialize_graphs` could be called (or some other new function, which would be a subpart of initialize graphs)

Reply:
> So I'm saying, maybe instead initialize_graphs could be called (or some other new function, which would be a subpart of initialize graphs)

I don't understand. `_safe_set_spin_projections` is called from `initialize_graphs` itself.

Reply:
Hmm, why i can't see this change anymore? The old code seems to use it the way I was suggesting

Reply:
7d780ec3